### ISSUE TITLE
Python problem in Mac os 10.15.

### ISSUE THREAD

**Issue by dr-costas** [vim/vim#6104]
**Describe the bug**
It seems that there is a big when using Python from Vim, with Vim 8.2, Python 3.7, and macOS 10.15

I try to use `:py3 pass` or `:py3 import vim` and I always get: 

````
Vim: Caught deadly signal ABRT
Vim: Finished

Abort trap: 6
````

I have nothing in `.vim`, I actually deleted the `.vim` directory. Still, there is the same error.  Even with `vim --clean`, again the same. 

**To Reproduce**
I compiled vim with: 

````
./configure --with-features=huge \
	   --enable-multibyte \
	   --enable-rubyinterp=yes \
	   --enable-python3interp=yes \
	   --with-python3-config-dir=“~/opt/anaconda3/lib/python3.7/config-3.7m-darwin/“ \
	   --enable-perlinterp=yes \
	   --enable-luainterp=yes \
	   --enable-cscope \
	   --prefix=/usr/local \
	   --with-vim-name=vim.8.2 \
	   --enable-autoservername \
	   --enable-terminal \
	   --enable-cscope
````

Then, I open Vim and I do: 

````
:py3 pass
````

or 

````
:py3 import vim
````

Vim exits with the abort trap message. 

**Environment (please complete the following information):**
 - Vim version: 

````
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled May 18 2020 18:34:53)
macOS version
Included patches: 1-788
Compiled by <me>@iMac
Huge version without GUI.  Features included (+) or not (-):
+acl               -farsi             +mouse_sgr         +tag_binary
+arabic            +file_in_path      -mouse_sysmouse    -tag_old_static
+autocmd           +find_in_path      +mouse_urxvt       -tag_any_white
+autochdir         +float             +mouse_xterm       -tcl
+autoservername    +folding           +multi_byte        +termguicolors
-balloon_eval      -footer            +multi_lang        +terminal
+balloon_eval_term +fork()            -mzscheme          +terminfo
-browse            -gettext           +netbeans_intg     +termresponse
++builtin_terms    -hangul_input      +num64             +textobjects
+byte_offset       +iconv             +packages          +textprop
+channel           +insert_expand     +path_extra        +timers
+cindent           +ipv6              +perl              +title
-clientserver      +job               +persistent_undo   -toolbar
+clipboard         +jumplist          +popupwin          +user_commands
+cmdline_compl     +keymap            +postscript        +vartabs
+cmdline_hist      +lambda            +printer           +vertsplit
+cmdline_info      +langmap           +profile           +virtualedit
+comments          +libcall           -python            +visual
+conceal           +linebreak         +python3           +visualextra
+cryptv            +lispindent        +quickfix          +viminfo
+cscope            +listcmds          +reltime           +vreplace
+cursorbind        +localmap          +rightleft         +wildignore
+cursorshape       -lua               +ruby              +wildmenu
+dialog_con        +menu              +scrollbind        +windows
+diff              +mksession         +signs             +writebackup
+digraphs          +modify_fname      +smartindent       -X11
-dnd               +mouse             -sound             -xfontset
-ebcdic            -mouseshape        +spell             -xim
+emacs_tags        +mouse_dec         +startuptime       -xpm
+eval              -mouse_gpm         +statusline        -xsmp
+ex_extra          -mouse_jsbterm     -sun_workshop      -xterm_clipboard
+extra_search      +mouse_netterm     +syntax            -xterm_save
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/local/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H   -DMACOS_X -DMACOS_X_DARWIN  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1       
Linking: gcc   -L. -L/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.Internal.sdk/usr/local/lib  -L/usr/local/lib -o vim.8.2        -lm -lncurses  -liconv -framework AppKit   -fstack-protector  -L/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE -lperl  -L~/opt/anaconda3/lib/python3.7/config-3.7m-darwin/ -lpython3.7m -framework CoreFoundation  -lruby.2.6  
````

 - OS: macOS 10.15.4
 - Terminal: bash

**Comment by dr-costas** [comment]
I tried also compiling vim with: 

````
./configure --with-features=huge \
	   --enable-multibyte \
	   --enable-rubyinterp=yes \
	   --enable-python3interp=yes \
	   --enable-perlinterp=yes \
	   --enable-luainterp=yes \
	   --enable-cscope \
	   --prefix=/usr/local \
	   --with-vim-name=vim.8.2 \
	   --enable-autoservername \
	   --enable-terminal \
	   --enable-cscope \
	   --with-python3-command=“~/opt/anaconda3/bin/python”
````

And without the `--with-python3-command`. For each of the different configurations, I tried the `vim --clean` and again, when using `:py3 pass` I got the `Abort trap 6`. 

I used with making `alias python=~/opt/anaconda3/bin/python`, again the same. 

I also tried with deactivating first the conda environment and then compiling, again the same. 

Finally, just to be sure, I repeated all the above but each time I deleted the `~vim` directory, re-cloned it from GitHub, run the config, and then compiled. The result was again `Abort trap 6`.

Any solutions?

**Comment by dr-costas** [comment]
Anyone who can help? Or, any pointer so I can search what is happening with my Vim?

**Comment by chrisbra** [comment]
This sounds like a library conflict. Try without anaconda first and check if this works. Those other python distributions have been source of trouble before. Also, you might want to try compiling that python from source, so that vim can pick up the right libraries that are actually used.

**Comment by rmax** [comment]
After a plugins update my vim freeze when editing any file. I was able to pin down to ultisnips and I fixed it by using `3.2` tag rather than latest master. I'm unsure why it freezes, though.

My `vim-plug` config:
```vim
Plug 'SirVer/ultisnips', {'tag': '3.2'} | Plug 'honza/vim-snippets'
```

Edit: my system is OSX and installed vim via brew (bottled 8.2 version).

**Comment by chrisbra** [comment]
that is a different problem so.
